broker/src/twilio/twilio_broker.erl

```diff
@@ -32,11 +36,36 @@ dispatch(_Session = #session{session_id = SessionId, channel = Channel, address
   ],
 
   Response = uri:post_form(RequestBody, [{basic_auth, {AccountSid, AuthToken}}], RequestUrl),
-  io:format("Response: ~p~n", [Response]),
+  lager:info("Twilio response: ~p~n", [Response]),
   case Response of
     {ok, {{_, 201, _}, _, _}} -> ok;
-    {ok, {{_, _, Reason}, _, _}} -> {error, Reason};
+    {ok, {{_, _, Reason}, _, Msg}} -> {error, parse_exception(Reason, Msg)};
     _ ->
       timer:apply_after(timer:minutes(1), broker, notify_ready, [?MODULE]),
       {error, unavailable}
   end.
+
+parse_exception(Reason, Message) ->
+  try
+    {#xmlElement{content = [RestException]}, _} = xmerl_scan:string(Message),
+    case RestException#xmlElement.name of
+      'RestException' ->
+        Content = RestException#xmlElement.content,
+        FullCode = case extract_exception_item('Code', Content) of
+          undefined -> undefined;
+          Code -> "twilio:" ++ Code
+        end,
+        {error, extract_exception_item('Message', Content), FullCode};
+      _ -> Reason
+    end
+  catch
+    _ -> Reason
+  end.
```

Comment:
I think it's probably better to use `xmerl_xs` here. See http://erlang.org/doc/man/xmerl_xs.html

Something like:

```erl
parse_exception(Reason, Message) ->
  try
    {Doc, []} = xmerl_scan:string(Message),
    [Exception] = xmerl_xs:select("/TwilioResponse/RestException", Doc),
    FullCode = case xmerl_xs:value_of(xmerl_xs:select("./Code", Exception)) of
      [Code] -> "twilio" ++ Code;
      _ -> undefined
    end,
    [Message] = xmerl_xs:value_of(xmerl_xs:select("./Message", Exception)),
    {error, Message, FullCode}
  catch
    _ -> Reason
  end.
```